torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -830,6 +830,113 @@ void removeMaxPoolUnusedOutput(Block* b) {
   }
 }
 
+// This optimization fuses LogSoftmax and NegativeLogLikelihoodLoss operators
+// into one operator: SoftmaxCrossEntropyLoss, and depending on the dimensions
+// of the input and different attributes there will be different subgraphs of
+// LogSoftmax and NegativeLogLikelihoodLoss.
+static void fuseLogSoftmaxNllLoss(Block* b) {
+  for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+    for (auto* child_block : it->blocks()) {
+      fuseLogSoftmaxNllLoss(child_block);
+    }
+    if (it->kind() == onnx::NegativeLogLikelihoodLoss) {
+      auto prev = it->input(0)->node();
+      Node* origNllLossNode = *it;
+      Node* origLogSoftmaxNode;
+      // if the input is 2D
```

Comment:
maybe we can move the comment to the first line inside each if-else block, instead of above each block.